Formula/p/pivit.rb

```diff
@@ -0,0 +1,18 @@
+class Pivit < Formula
+  desc "Sign and verify data using hardware (Yubikey) backed x509 certificates (PIV)"
+  homepage "https://github.com/cashapp/pivit"
+  url "https://github.com/cashapp/pivit/archive/refs/tags/v0.6.0.tar.gz"
+  sha256 "67a2f353c820aad9c9a4cc937ba256c75cb36044baa0578b74a1bfbdf4e6bd90"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "make", "pivit"
+    bin.install "pivit"
+  end
+
+  test do
+    system "make", "test"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for the details explanation! It took me a while to go through the fine details in the cookbook to see get to these parts :)

This tool requires a Yubikey (or other similar hardware) to be present in order to use it.
I updated the test to call `pivit --print` which should try and load a certificate from the hardware backed key, and print it.
The output of that command will either be a PEM encoded certificate, or a message specifying either the certificate or the hardware weren't found.
In either case, it'll exercise the more relevant code paths to make sure it is operational.